spec/presenters/new_metrics_presenter_spec.rb

```diff
@@ -0,0 +1,287 @@
+require "rails_helper"
+
+RSpec.describe NewMetricsPresenter do
+  let(:prison) { create(:prison) }
+  let(:other_prison) { create(:prison) }
+
+  def refresh_views
+    VisitCountsByPrisonStateDateAndTimely.refresh
+    PercentilesByCalendarDate.refresh
+    PercentilesByPrisonAndCalendarDate.refresh
+    RejectionPercentageByDay.refresh
+  end
+
+  describe 'Percentiles Statistics' do
+    let(:visits) { create_list(:visit, 10, prison: prison) }
+    before do
+      travel_to(visits.first.created_at + 2.days) do
+        visits[0..8].each do |visit|
+          visit.assign_attributes(slot_granted: visit.slots.first)
+          BookingResponder::Accept.new(BookingResponse.new(visit: visit)).process_request
+        end
+      end
+
+      travel_to(visits.first.created_at + 6.days) do
+        visit = visits.last
+        visit.assign_attributes(slot_granted: visit.slots.first)
+        BookingResponder::Accept.new(
+          BookingResponse.new(visit: visit)
+        ).process_request
+      end
+
+      refresh_views
+    end
+
+    describe '#percentiles_by_day' do
+      it 'returns the distributions of time to process a request' do
+        nineteenth = subject.percentiles_by_day.first.percentiles.first / 86_400
+        median     = subject.percentiles_by_day.first.percentiles.last / 86_400
+        expect(nineteenth).to eq(5)
+        expect(median).to be_within(1).of(1)
+      end
+    end
+
+    describe '#percentiles_by_day_for' do
+      it 'returns the distributions of time to process a request for a given prison' do
+        nineteenth = subject.percentiles_by_day_for(prison).first.percentiles.first / 86_400
+        median     = subject.percentiles_by_day_for(prison).first.percentiles.last / 86_400
+        expect(nineteenth).to be_within(1).of(5)
+        expect(median).to be_within(1).of(1)
```

Comment:
Can we have more precision with these `be_within` ? The possible acceptable range is pretty big given the expected result